photon-lib/src/main/java/org/photonvision/RobotPoseEstimator.java

```diff
@@ -0,0 +1,139 @@
+package org.photonvision;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.photonvision.targeting.PhotonTrackedTarget;
+
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.wpilibj.DriverStation;
+
+public class RobotPoseEstimator {
+
+    /**
+     * <ul>
+     * <li><strong>LOWEST_AMBIGUITY</strong>: Choose the Pose with the lowest ambiguity</li>
+     * <li><strong>CLOSEST_TO_CAMERA_HEIGHT</strong>: Choose the Pose which is closest to the camera height</li>
+     * <li><strong>CLOSEST_TO_REFERENCE_POSE</strong>: Choose the Pose which is closest to the pose from setReferencePose()</li>
+     * </ul>
+     */
+    enum PoseStrategy {
+        LOWEST_AMBIGUITY, // TODO: Test
+        CLOSEST_TO_CAMERA_HEIGHT, // TODO: Implement
+        CLOSEST_TO_REFERENCE_POSE // TODO: Implement
+    
+    }
+    
+    
+    private Map<Integer, Pose3d> aprilTags;
+    private PoseStrategy strategy;
+    private ArrayList<Pair<PhotonCamera, Transform3d>> cameras;
+    private Pose3d lastPose;
+    
+    /**
+     * Create a new RobotPoseEstimator. 
+     * <p>
+     * Example: 
+     *  <code>
+     *  <p>
+     *  Map<Integer, Pose3d> map = new HashMap<>();
+     *  <p>
+     *  map.put(1, new Pose3d(1.0, 2.0, 3.0, new Rotation3d())); // Tag ID 1 is at (1.0,2.0,3.0)
+     *  </code>
+     * 
+     * @param aprilTags A Map linking AprilTag IDs to Pose3ds with respect to the FIRST field.
+     * @param strategy The strategy it should use to determine the best pose.
+     * @param cameras An ArrayList of Pairs of PhotonCameras and their respective Transform3ds from the center of the robot to the cameras. 
+     */
+    public RobotPoseEstimator(Map<Integer, Pose3d> aprilTags, PoseStrategy strategy, ArrayList<Pair<PhotonCamera, Transform3d>> cameras) {
+        this.aprilTags = aprilTags;
+        this.strategy = strategy;
+        this.cameras = cameras;
+    }
+
+
+    public Pose3d update() {
+        if (cameras.isEmpty()) {
```

Comment:
If so I'd make it toggleable. I'm not well versed in using Field2D beyond the simple x,y. This would also require the map to be specified, right? Decreases game generality.

Reply:
It can accept many poses. The background is actually set on the user side, it just shows shapes at the x/y/theta coordinates that are specified.

Reply:
Re: toggleable.... possibly? That would follow pattern with wpilib's sendable, and reduce NT server load if folks aren't using it. However, it's still on the user to add whatever is published to some dashboard to view, so it shouldn't be "in the way" even if they turn it on.